soccer/planning/Trajectory.hpp

```diff
@@ -0,0 +1,422 @@
+#pragma once
+
+#include <Geometry2d/Pose.hpp>
+#include <time.hpp>
+
+#include "DebugDrawer.hpp"
+#include "Instant.hpp"
+#include "Utils.hpp"
+#include "planning/DynamicObstacle.hpp"
+
+namespace Planning {
+
+/**
+ * @brief An ordered sequence (by time) of RobotInstants
+ */
+using RobotInstantSequence = std::vector<RobotInstant>;
+
+/**
+ * @brief Represents a trajectory x(t), y(t), h(t), with a smooth velocity and
+ * piecewise-continuous acceleration.
+ *
+ * @details Note that this provides two interfaces: one accepting RJ::Seconds,
+ * which is a duration, and one accepting RJ::Time, an instant in time.
+ *
+ * This uses cubic interpolation between two adjacent instants, fitting a
+ * parametric cubic polynomial based on initial and final posiiton and velocity.
+ *
+ * A trajectory also carries a timestamp representing the time at which it was
+ * planned. This should be filled in using `trajectory.stamp(RJ::Time)` before
+ * publishing, and will be reset to `nullopt` when the trajectory is modified.
+ */
+class Trajectory {
+public:
+    Trajectory() = default;
+    ~Trajectory() = default;
+
+    /**
+     * @brief Create a trajectory from several "instants", each with a pose,
+     * velocity, and timestamp. This
+     */
+    explicit Trajectory(RobotInstantSequence instants)
```

Comment:
should use an r-value reference or a smart pointer in the function parameter. I think this makes a copy

Reply:
To pass into this you can do `Trajectory(std::move(instants_arg))` and it will move `instants_arg` into the `instants` argument (call the move constructor for the parameter. This is generally preferred because you can pass either a const reference or an rvalue.